internal/batch/batch_processor.go

```diff
@@ -633,6 +633,12 @@ func (bp *batchProcessor) dispatchBatch(payload *DispatchPayload) error {
 						}
 					}
 				}
+				conflictErr, conflictTestOk := err.(operations.ConflictError)
+				if conflictTestOk && conflictErr.IsConflictError() {
+					// We know that the connector has received our batch, so we shouldn't need to retry
+					payload.addMessageUpdate(payload.Messages, core.MessageStateReady, core.MessageStateSent)
```

Comment:
I think we need the whole block and we can extract this to a function
```Golang
if core.IsPinned(payload.Batch.TX.Type) {
   payload.addMessageUpdate(payload.Messages, core.MessageStateReady, core.MessageStateSent)
} else {
   payload.addMessageUpdate(payload.Messages, core.MessageStateReady, core.MessageStateConfirmed)
}
```
For pinned batches we have to wait for the blockchain TX that is why we set them as sent and you might have just assumed this was always the case in conflict mode. But this batch processor is completed agnostic to how the batch is handled, it could be a conflict from data exchange or any other plugin needed to dispatch that batch. In the case where it's not pinned, we don't need to wait for TX so we can set it as confirmed

Reply:
Furthermore there is a snippet of code above that handles a cancelled batch pin and this is an edge case but if the last error before the cancel is a conflict we will update the state of the initial batch incorrectly from cancelled to sent